src/librustc_mir_build/build/expr/as_operand.rs

```diff
@@ -21,6 +21,37 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         self.as_operand(block, local_scope, expr)
     }
 
+    /// Returns an operand suitable for use until the end of the current
+    /// scope expression and suitable also to be passed as function arguments.
+    ///
+    /// The operand returned from this function will *not be valid* after
+    /// an ExprKind::Scope is passed, so please do *not* return it from
+    /// functions to avoid bad miscompiles.
+    crate fn as_call_temporary<M>(&mut self, block: BasicBlock, expr: M) -> BlockAnd<Operand<'tcx>>
+    where
+        M: Mirror<'tcx, Output = Expr<'tcx>>,
+    {
+        let local_scope = self.local_scope();
+        let expr = self.hir.mirror(expr);
+
+        if let ExprKind::Deref { ref arg } = expr.kind {
+            let tcx = self.hir.tcx();
+            let param_env = self.hir.param_env;
+
+            let arg = self.hir.mirror(arg.clone());
+            let ty = arg.ty;
+            let span = arg.span;
+
+            if !ty.is_sized(tcx.at(span), param_env)
+                && !ty.is_copy_modulo_regions(tcx, param_env, span)
```

Comment:
what does `!Copy` have to do with anything here?

Reply:
I'm guessing this is addressing

> The argument is moved (not copied),

I don't understand this reasoning, but note that I have not read the issue at all in the last weeks

Reply:
And it's redundant with `!is_sized`, because `Copy` requires `Sized` today.

Reply:
Maybe add `assert!(!ty.is_copy_modulo_regions(tcx, param_env, span))` with a comment about this being an unsized move.

Reply:
Done!